File: C/StandardAlgo/Reculsive/GCD/gcd.c
#include<stdio.h>
int gcd(int, int);
int gcd2(int, int);
int main(void) {
	printf("%d\n", gcd(10, 8));
	printf("%d\n", gcd2(10, 5));
	return 0;
}
int gcd(int m, int n) {
	if (n == 0)return m;
	else gcd(n, m % n);
}
int gcd2(int m, int n) {
	if (m < n) {
		int temp = m; m = n; n = temp;
	}
	if (m % n == 0)
		return n;
	else {
		gcd2(n, m % n);
	}
}

File: Java/StandardAlgo/Algorithm/src/recursion/GCD.java
package recursion;

public class GCD {
	public static void main(String[] args) {
		System.out.println(gcd(10,5));
		System.out.println(gcd2(5,10));
	}
	public static double gcd(int m, int n) {
		if(m<n){
			int tmp=m; m=n; n=tmp;
		}
		if(m%n==0) {
			return n;
		}else {
			return gcd(n, m%n);
		}
	}
	public static double gcd2(int p, int q) {
		if(q==0)
			return p;
		else
			return gcd2(q, p%q);
	}
}
